File: CucumberJVM/Chapter15/15.7-StubDependencyPattern/After/src/test/java/com/wimp/specs/support/PlaceOrderRequestObjectMother.java
package com.wimp.specs.support;

import com.wimp.app.restapi.PlaceOrderRequest;

import java.time.LocalDateTime;

public class PlaceOrderRequestObjectMother {
    private LocalDateTime expectedDeliveryTime;

    public PlaceOrderRequest build() {
        return new PlaceOrderRequest(expectedDeliveryTime);
    }

    public PlaceOrderRequestObjectMother withExpectedDeliveryTime(LocalDateTime time) {
        this.expectedDeliveryTime = time;
        return this;
    }
}

File: Reqnroll/Chapter17/17.2-InPlacePersonaPattern/After/WIMP.Specs/Support/PlaceOrderRequestObjectMother.cs
using WIMP.App.Models;
using WIMP.App.RestApi;

namespace WIMP.Specs.Support;

public class PlaceOrderRequestObjectMother
{
    private readonly List<PizzaItem> items = [];
    private readonly string email = DomainDefaults.CustomerEmail;
    private readonly string deliveryAddress = DomainDefaults.CustomerAddress;
    private DateTimeOffset? expectedDeliveryTime;

    public PlaceOrderRequestObjectMother()
    {
        items.Add(DomainDefaults.PizzaItemDefaultInstance());
    }

    public PlaceOrderRequest Build()
    {
        return new(
            items.ToArray(),
            deliveryAddress,
            email,
            ExpectedDeliveryTime: expectedDeliveryTime);
    }

    public PlaceOrderRequestObjectMother WithExpectedDeliveryTime(DateTimeOffset time)
    {
        expectedDeliveryTime = time;
        return this;
    }
}
